### DISCUSSION TITLE
关于panjf2000 / ants v2.0.0版本修复的问题

### DISCUSSION THREAD

**Discussion by guoxinyu-shopee** [panjf2000/ants#162] [Q&A]
你好，我在v2.0.0看到如下描述
![image](https://user-images.githubusercontent.com/53163548/120151300-a9e38600-c21e-11eb-93c3-078fb0ad358b.png)

但是对比v1.2.0和v2.0.0的代码，我没看懂是如何解决这个问题的，或者这个问题是什么情况下会出现？
可否给我讲解一下？

**Comment by panjf2000** [comment]
定时任务把所有 workers 都清理了，但是还有一些调用方在等待唤醒信号，这样会永远阻塞。

**Reply by guoxinyu-shopee** [reply]
> 定时任务把所有 workers 都清理了，但是还有一些调用方在等待唤醒信号，这样会永远阻塞。

那是哪段代码解决了这个问题？
在v2.0.0中谁来唤醒等待唤醒的协程呢？

**Reply by guoxinyu-shopee** [reply]
> 定时任务把所有 workers 都清理了，但是还有一些调用方在等待唤醒信号，这样会永远阻塞。

这个场景我理解到了，但是目前还不清楚，是如何解决的？
我看在1.2.0版本在定时清理任务中，已经有如下代码了
if p.Running() == 0 {
   p.cond.Broadcast()
}

**Comment by guoxinyu-shopee** [comment]
这个场景我理解到了，但是目前还不清楚，是如何解决的？
我看在1.2.0版本在定时清理任务中，已经有如下代码了

if p.Running() == 0 {
   p.cond.Broadcast()
}


Andy Pan ***@***.***> 于2021年5月31日周一 下午2:53写道：

**Comment by panjf2000** [comment]
看提交的时间应该是重复了，在 v1.2.0 的时候就修复了。
